examples/widgets/annotated_cursor.py

```diff
@@ -0,0 +1,345 @@
+            # if the returned plot point is valid
+            if plotpoint is not None:
+                event.xdata = plotpoint[0]
+                event.ydata = plotpoint[1]
+
+        # If the plotpoint is given, compare to last drawn plotpoint and
+        # return if they are the same.
+        # Skip even the call of the base class, because this would restore the
+        # background, draw the cursor lines and would leave us the job to
+        # re-draw the text.
+        if plotpoint is not None:
+            if plotpoint == self.lastdrawnplotpoint:
+                return
+
+        # Baseclass redraws canvas and cursor. Due to blitting,
+        # the added text is removed in this call, because the
+        # backround is redrawn.
+        super().onmove(event)
+
+        # Check if the display of text is still necessary.
+        # If not, just return.
+        # This behaviour is also cloned from the base class.
+        if (not self.get_active()) or (not self.visible):
+            return
+
+        # Draw the widget, if event coordinates are valid
+        # If plotpoint is valid (not None)
+        if plotpoint is not None:
+            # Update position and displayed text.
+            # Position: Where the event occured.
+            # Text: Determined by setpos() method earlier
+            # Position is transformed to pixel coordinates,
+            # an offset is added there and this is transformed back.
+            temp = [event.xdata, event.ydata]
+            temp = self.ax.transData.transform(temp)
+            temp = temp + self.offset
+            temp = self.ax.transData.inverted().transform(temp)
+            self.text.set_position(temp)
+            self.text.set_text(self.numberformat.format(*plotpoint))
+            self.text.set_visible(self.visible)
+
+            # Tell base class, that we have drawn something.
+            # Baseclass needs to know, that it needs to restore a clean
+            # backround, if the cursor leaves our figure context.
+            self.needclear = True
+
+            # Remember the recently drawn cursor position, so events for the
+            # same position (mouse moves slightly between two plot points)
+            # can be skipped
+            self.lastdrawnplotpoint = plotpoint
+        # otherwise, make text invisible
+        else:
+            self.text.set_visible(False)
+
+        # Draw changes. Canot use _update method of baseclass,
+        # because it would first restore the backround, which
+        # is done already and is not necessary.
+        if self.useblit:
+            self.ax.draw_artist(self.text)
+            self.canvas.blit(self.ax.bbox)
+        else:
+            # If blitting is deactivated, the overwritten _update call made
+            # by the base class immedeately returned.
+            # We still have to draw the changes.
+            self.canvas.draw_idle()
+
+    def setpos(self, xpos, ypos):
```

Comment:
```suggestion
    def set_position(self, xpos, ypos):
```

Reply:
Fixed in fee740e298d6a9bb863dd452900c43b8cc97c98f in multiple spots